### ISSUE TITLE
Chart is taking time to zoom in mobile view

### ISSUE THREAD

**Issue by krupali-clavier** [highcharts/highcharts#19785]
#### Expected behaviour
https://jsfiddle.net/KrupaliClavier/7vea0Lfg/4/ chart this is our web url and we need this working same as web view

#### Actual behaviour
When we're trying to use this url in our web-view of react-native behave it's taking time to zoom in-out. 

#### Live demo with steps to reproduce
https://jsfiddle.net/KrupaliClavier/7vea0Lfg/4/

#### Product version
latest

#### Affected browser(s)
App (react-native)

**Comment by karolkolodziej** [comment]
Hi @krupali-clavier!

Can you please elaborate on where the difference between mobile/web zooming time is?
I don't see any difference in the zooming performance.

**Comment by krupali-clavier** [comment]
We're having a problem when we try to zoom in on an app by using a web-view
of a specific URL. Please find an attachment.

Thank you

**Comment by karolkolodziej** [comment]
Yes, I can see the `Chart Steaming` you have provided but I still don't understand the problem you are having with it.
Could you please explain?

**Comment by krupali-clavier** [comment]
We've an app for showing the chart of trading and showing charts using this
web-url by WEB-VIEW of react-native (http://139.59.106.6:3000/chart) and
when we try to zoom in the app, we face this kind of issue.

**Comment by karolkolodziej** [comment]
Your chart from the `Chart Stream` looks and behaves differently. 
There are a lot of updates, are they necessary? How are they performed? Can you walk me through the changes between mobile/web?

**Comment by krupali-clavier** [comment]
https://jsfiddle.net/KrupaliClavier/7vea0Lfg/4/
This is our chart that should be working like this but it's working in an
app like the video attached. Please have a look into that. we're using
Web-view from react-native.

**Comment by karolkolodziej** [comment]
There is a difference between them. Are you sure that there are no additional updates/redraws?
Can you share the code for React Native?

**Comment by krupali-clavier** [comment]
I'm sharing the code of react-native . Please check
`import React, {useState} from 'react';
import {WebView} from 'react-native-webview';
import {FullScreenLoader} from './LoaderScreen';

const URL = 'http://139.59.106.6:3000/chart';
export const WebChartView = ({style}) => {
  const [loader, setLoader] = useState(true);
  return (
    <>
      <WebView
        useWebKit={true}
        javaScriptEnabled={true}
        source={{uri: URL}}
        style={style}
        scalesPageToFit={true}
        scrollEnabled={true}
        nestedScrollEnabled={true}
        decelerationRate={'fast'}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        onLoadProgress={({nativeEvent}) => {
          const progress = nativeEvent.progress;
          if (progress > 0.8) setLoader(false);
        }}
        onError={() => setLoader(false)}
      />
      {loader && <FullScreenLoader />}
    </>
  );
};`

**Comment by h4rshil** [comment]
@karolkolodziej it is just the webview on mobile as well so it should work as it is working on mobile browser right? but on mobile browser also zoom becomes laggy

**Comment by bm64** [comment]
Hi @krupali-clavier @h4rshil 

I want to clarify that WebView essentially renders web content as-is. This means if the original web page (http://139.59.106.6:3000/chart) has performance issues, those will carry over into the React Native WebView environment.

It appears there's a discrepancy between your live chart and the static JSFiddle demo. The live chart seems dynamic with frequent updates, while the JSFiddle is static. This makes it challenging to provide targeted solutions without seeing the chart's underlying code.

My suggestion would be to first focus on optimizing the chart rendered at your web URL. This will inherently improve the WebView performance. To give specific guidance, we'd need to examine the chart's code to suggest optimizations.

**Comment by krupali-clavier** [comment]
Hello Highcharts/Highcharts,
                    We've checked the live chart as per our URL, it's
working fine by url.. If you need to check code of that Please check in
attachment, I've attached a file with code.

Thank you

**Comment by bm64** [comment]
@krupali-clavier  I've checked the code and the live chart from the URL. There's nothing inherently incorrect or buggy with Highcharts' behavior. The performance issues are a result of too many updates and unoptimized code on your end. Specifically, functions like `addPoint` and `update` in the socket's events.load, and `setData` in setExtremes are triggering automatic redraws, causing the slowdown.

To improve performance, use `setData()` and `update()` with the redraw flag set to false, then consolidate all updates with a single chart.redraw() at the end of your socket events. It might also be beneficial to split your single socket into smaller, specialized events to reduce the update frequency.
API reference:
https://api.highcharts.com/class-reference/Highcharts.Series#setData
https://api.highcharts.com/class-reference/Highcharts.Point#update
https://api.highcharts.com/class-reference/Highcharts.Axis#update

As a note, the live chart from the URL does exhibit performance issues on desktop browsers. I've tested it on an iPhone 7 and it was non-interactive, suggesting even more significant performance issues on mobile devices.

**Comment by krupali-clavier** [comment]
Hello Mateusz,
                We've tried this method as you've suggested, it's improved
but not performing well as expected. Please guide.

**Comment by bm64** [comment]
@krupali-clavier please provide updated code and a working chart if possible. The update frequency is driven by your socket events, so if it updates are too frequent, this can lead to performance issues regardless of the chart config.

**Comment by krupali-clavier** [comment]
I'm giving you the latest code of the chart we applied. Please verify and
guide us what we can change to improve performance.

**Comment by krupali-clavier** [comment]
Hello Highcharts/Highcharts,
                          We're waiting for your reply as you asked for
latest code, we've attached. Please look into it and get back to us as soon
as possible.


Thank you

**Comment by bm64** [comment]
@krupali-clavier I couldn't find any code in your reply. Could you provide me with a jsfiddle demo like you did before? https://jsfiddle.net/KrupaliClavier/7vea0Lfg/4/ Also it might be necessary to make the socket updates work in the demo, so that I know how frequently the chart updates. Without that I might not be able to help.

**Comment by bm64** [comment]
@krupali-clavier please try changes from this demo: https://jsfiddle.net/BlackLabel/3opzyfg5/
The changes are located in `load event`, in `socket.on` function. Let me know if the chart still works correctly and if the performance is better.

**Comment by krupali-clavier** [comment]
Hi Mateusz,
               Thanks for the response but this is not working for me, it
doesn't improve the performance of our chart. Please review the code from
the attached file and guide us. This is our URL of the chart showing.
http://139.59.106.6:3000/chart

**Comment by KacperMadej** [comment]
Hi @krupali-clavier 

I have checked the code you have available on the shared URL - the one that you have provided.
Your socket triggers a lot of non-optimized updates that trigger many chart redraws. As mentioned already in this issue thread the updates should be done with the redraw flag set to false. In case you are unable to confirm if a function call supports a redraw argument please refer to our API documentation.

To improve performance you could disable all animation - both in the chart's configuration (chart and series animation-related options can be checked in the API reference - https://api.highcharts.com/highcharts/chart.animation, https://api.highcharts.com/highcharts/plotOptions.series.animation) as well as when using the dynamic update functions.

The site you have shared via the URL is very slow on a PC with more memory available than a standard mobile device. That suggests the overall performance of your code is below expectations regardless of the used machine or browser. You could improve the performance by reducing the frequency of the socket interactions and the updates triggered through the socket. For example, if there is no change then don't update the element that is scheduled for update in your code at this moment - the title and subtitle are updated every second (every time the socket gets a response and is processing the new data). If there is no change do not fire update functions or any other functions that are not needed to improve the performance.

You are calling an axis update that changes the title (and most of the time, if not all the time, the text is not changed at all) but that triggers the redraw of most of the chart elements that are destroyed or at least recalculated. The performance will be improved if the amount of not needed function calls and updates is reduced.

Please reduce the socket frequency (at least temporarily for the sake of debugging and testing performance) because your website runs very slow after a few minutes of interacting with it and I am not even able to run any meaningful performance or memory checks. The number of updates and redraws could be reduced to update the chart after a few socket connections so there is enough time in between the updates to keep the website responsive.

Please note that your replies are publicly available on GitHub's issue thread (https://github.com/highcharts/highcharts/issues/19785) and no attachments are delivered there so if you would like to share the code please use a URL or a live demo (e.g. JSFiddle).

I understand your urgency for the solution as the conversation in this thread has continued for many weeks already so in case you have the advantage license active you could reach our technical support during our working hours for a faster reply or smoother conversation - https://www.highcharts.com/support/. More information could be found in the linked website and the chat is located in the bottom right corner. This being said it would be great if we could continue the conversation here to keep all the info in the same place for easier access.
